main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	mapset "github.com/deckarep/golang-set"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+)
+
+func main() {
+	start()
+}
+
+func start() {
+	f, err := os.Open("TestExpectations.txt")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		expectation := scanner.Text()
+		url, tags, testname, results := parseExpectation(expectation)
+		if isMeetCriteria(url, tags, testname, results) {
+			bsfSet := getBSF()
```

Comment:
Not that it matters, but is this opening the BSF file, reading it, and discarding it for every matching line in TestExpectations?